src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs

```diff
@@ -284,7 +284,29 @@ private void ChangeProjectProperty<T>(
 
                     if (!isFullyLoaded)
                     {
-                        TryReportCompilationThrownAway(_projectSystemProjectFactory.Workspace.CurrentSolution, Id);
+                        var currentSolution = _projectSystemProjectFactory.Workspace.CurrentSolution;
+                        var reportCompilationThrownAway = true;
+
+                        // Preprocessor directive only ParseOptions changes have special handling in DocumentState.UpdateParseOptionsAndSourceCodeKind.
+                        // We don't want to report telemetry for those changes or for the initial evaluation result.
+                        if (newValue is ParseOptions newParseOption)
+                        {
+                            if (oldValue is ParseOptions oldParseOptions)
+                            {
+                                var projectState = currentSolution.SolutionState.GetRequiredProjectState(Id);
```

Comment:
Rather than fetching the project to get language services, you should be able to shortcut and just do _projectSystemProjectFactory.SolutionServices.GetRequiredLanguageService<ISyntaxTreeFactoryService>(Language). That use of CurrentSolution should be fine, but this is probably shorter and doesn't require me to think about it.

Reply:
Yeah, that's nicer.